### DISCUSSION TITLE
Making a class diagram using mermaid.js

### DISCUSSION THREAD

**Discussion by ghost** [YuriSizov/boscaceoil-blue#13] [Show and tell]
Hello. I have forked your project and started making a UML Class Diagram taking in count only _.gd_ files.
You can look from my fork an example by starting making a class diagram.

[boscaceoil-blue-class-diagram](https://github.com/SimoneStarace/boscaceoil-blue/blob/uml_diagrams/uml/class.md)

Are you interested in this?

- If yes then I will continue updating the diagram.
- If not then I'll stop working on it.

Please let me know.
Have a nice day.

**Comment by YuriSizov** [comment]
Hey there! Thanks for your interest in contributing to the project, but may I ask how this would be used and what is this going to help with? Like, what is the problem that you're trying to solve here?

**Comment by ghost** [comment]
Hi @YuriSizov 
Thanks for answering.
> but may I ask how this would be used and what is this going to help with? Like, what is the problem that you're trying to solve here?

Basically a Class Diagram can make an overview of which classes are used or inherith from which one, basically the relationship between GDScripts in this case.
It could be helpful to understand which classic must be taken in count when the software must be modified.

Let me know.

**Reply by YuriSizov** [reply]
Right, yes, I understand what a class diagram itself is. I just want to understand why in your opinion it's something that we need right now. Class dependencies are pretty explicit in GDScript, and I can't think of a case where you'd need to look for extra classes to take into account while making changes. Everything should be pretty obvious from the code alone.

Also, and correct me if I'm wrong, but you're not making this diagram use any form of automation, do you? I assume so from your opening message which makes it look like a rather laborious process. If that's the case and the diagram cannot be updated automatically, then its usefulness is going to be limited. All documentation for the code that is node created by that code is going to become outdated at some point. At best, we'd have to put in extra work to remind ourselves and other contributors to keep this diagram up-to-date. At worst, people will find an outdated version and assume relationships wrongly, get confused.

So, in short, things need to be done for a specific purpose and without creating extra labor and maintenance cost for contributors.

**Reply by ghost** [reply]
Hallo. 
Thank for answering.
> I just want to understand why in your opinion it's something that we need right now. Class dependencies are pretty explicit in GDScript, and I can't think of a case where you'd need to look for extra classes to take into account while making changes. Everything should be pretty obvious from the code alone.

You are corrent about the classes dependencies being esplicit since I was thinking the project would have many classes with dependencies etc. 

> and correct me if I'm wrong, but you're not making this diagram use any form of automation, do you?

Unfortunately it is all made manually. I'm using mermaid.js to make the diagram and it doesn't feature an automation process because it simply a javascript 'framework' for making diagrams.

> If that's the case and the diagram cannot be updated automatically, then its usefulness is going to be limited. All documentation for the code that is node created by that code is going to become outdated at some point. At best, we'd have to put in extra work to remind ourselves and other contributors to keep this diagram up-to-date.

As you wrote, updating the diagram must be made manually by a person and it adds more work to do. 

> So, in short, things need to be done for a specific purpose and without creating extra labor and maintenance cost for contributors.

Understood. 
Thank you again for this discussion. 
This discussion can be closed from my perspective.
